### DISCUSSION TITLE
source .env doesnt work

### DISCUSSION THREAD

**Discussion by RaferGuo** [Cyfrin/foundry-full-course-cu#1467] [Q&A] [answered]
when i print ehco private key, it shows blank, that means source .env didnt work. i read some other Q&A, but didnt  find right answer,so how can i fix it, thx.

**Comment by EngrPips** [comment, marked as answer]
Hello @RaferGuo , It happens that in your `.env` file you have the variable `PRIVTE_KEY` and in your terminal you were finding the variable `PRIVATE_KEY` , so since what you were finding is not actually defined in your `.env` file you got nothing , you can fix it by doing one of the two below 

- in your terminal type `echo $PRIVTE_KEY` and you would get back the private key 
- change the variable name in your `.env` from `PRIVTE_KEY` to `PRIVATE_KEY` and type `echo $PRIVATE_KEY` in your terminal
